frontend/src/app/libs/file_builder.cljs

```diff
@@ -262,12 +261,14 @@
     (uuid/next))
 
   (export [_]
-    (->> (export-file file)
-         (rx/subs!
-          (fn [value]
-            (when  (not (contains? value :type))
-              (let [[file export-blob] value]
-                (dom/trigger-download (:name file) export-blob))))))))
+    (js/Promise.
+     (fn [resolve _reject]
+       (-> (export-file file)
+           (rx/subscribe
+            (fn [value]
+              (when (not (contains? value :type))
+                (let [[_ export-blob] value]
+                  (resolve export-blob))))))))))
```

Comment:
A small improvement:

```clojure
;; on requires
[promesa.core :as p]


(p/create
 (fn [resolve reject]
   (-> (export-file file)
       (rx/take 1)
       (rx/subscribe
        (fn [value]
          (when (not (contains? value :type))
            (let [[_ export-blob] value]
              (resolve export-blob))))
        reject))))
```

Reply:
I tried your suggestion and got this error message when exporting:

```
VM733:13 Uncaught (in promise) TypeError: h.subscribe is not a function
    at hn._subscribe (<anonymous>:13:45556)
    at hn._trySubscribe (<anonymous>:13:2676)
    at hn.subscribe (<anonymous>:13:2620)
    at Wpn (<anonymous>:238:86370)
    at <anonymous>:347:34481
    at Gpn (<anonymous>:238:87279)
    at k.export (<anonymous>:347:34456)
    at E (<anonymous>:347:43512)
```

The stack trace probably won't be enough to debug...

Reply:
Created a MR, with your commit and the change https://github.com/penpot/penpot/pull/4802, if it is ok for you, I will merge it ASAP